File: tests/fixtures/real_world/java/state/branch_close.java
import java.io.*;

public class BranchClose {
    public void conditionalClose(String path, boolean flag) throws IOException {
        FileInputStream fis = new FileInputStream(path);
        if (flag) {
            byte[] data = new byte[1024];
            fis.read(data);
            fis.close();
        }
        // fis leaked if !flag
    }

    public void bothBranchesClose(String path, boolean flag) throws IOException {
        FileInputStream fis = new FileInputStream(path);
        if (flag) {
            byte[] data = new byte[1024];
            fis.read(data);
            fis.close();
        } else {
            fis.close();
        }
    }
}

File: tests/fixtures/real_world/go/state/branch_close.go
package main

import "os"

func branchLeak(path string, flag bool) {
	f, _ := os.Open(path)
	if flag {
		buf := make([]byte, 1024)
		f.Read(buf)
		f.Close()
	}
	// f leaked if !flag
}
